x/reports/genesis.go

```diff
@@ -0,0 +1,31 @@
+package reports
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/desmos-labs/desmos/x/posts"
+	abci "github.com/tendermint/tendermint/abci/types"
+)
+
+// ExportGenesis returns the GenesisState associated with the given context
+func ExportGenesis(ctx sdk.Context, k Keeper) GenesisState {
+	return GenesisState{
+		Reports: k.GetReportsMap(ctx),
+	}
+}
+
+// InitGenesis initializes the chain state based on the given GenesisState
+func InitGenesis(ctx sdk.Context, keeper Keeper, data GenesisState) []abci.ValidatorUpdate {
+	for postID, reports := range data.Reports {
+		postID := posts.PostID(postID)
+		if !postID.Valid() {
+			panic(fmt.Errorf("invalid postID: %s", postID))
+		}
```

Comment:
I think that it might be better here to use the `ParsePostID` function to make to compatible with future changes if there ever going to be: 

```suggestion
		postID, err := posts.ParsePostID(postID)
		if err != nil {
			panic(err)
		}
```